controllers/kyma_controller.go

```diff
@@ -104,6 +102,13 @@ func (r *KymaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return ctrl.Result{}, client.IgnoreNotFound(err) //nolint:wrapcheck
 	}
 
+	if status.SetConditions(kyma, r.WatcherEnabled(kyma)) {
```

Comment:
Please consider renaming it to `InitConditions`, set method is misleading here, the function only initializes conditions, it's not for generic set condition purposes.